### PR TITLE
refactor: support to convert time string to timestamp in `convert_value()`

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This PR supports converting the time string(for example, `2024-12-26 10:00:25.899`) to the `Timestamp` type in `convert_value()`.  It fixes the insert data error when users insert timestamp in string format by using a prepared statement. The logic is compatible with native MySQL.

## PR Checklist
Please convert it to a draft if some of the following conditions are not met.

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.
- [ ] API changes are backward compatible.
- [ ] Schema or data changes are backward compatible.

### CODE REVIEW

File: src/servers/src/mysql/helper.rs [hunk 1]
```diff
@@ -161,7 +163,28 @@ pub fn convert_value(param: &ParamValue, t: &ConcreteDataType) -> Result<ScalarV
...
+                        Ok(ScalarValue::TimestampNanosecond(Some(ts.value()), None))
+                    }
+                    TimestampType::Microsecond(_) => {
+                        Ok(ScalarValue::TimestampMicrosecond(Some(ts.value()), None))
+                    }
+                    TimestampType::Millisecond(_) => {
+                        Ok(ScalarValue::TimestampMillisecond(Some(ts.value()), None))
+                    }
+                    TimestampType::Second(_) => {
+                        Ok(ScalarValue::TimestampSecond(Some(ts.value()), None))
```
Comment (sunng87) [inline src/servers/src/mysql/helper.rs:184]: I'm afraid we cannot simply use `ts.value` here for different time unit. The input timestamp string will have no unit information so the timestamp we parsed in line 167 will have a fixed unit.

Comment (zyy17) [reply src/servers/src/mysql/helper.rs:184]: Maybe we can use `covert_to()`, for example:

```
let ts = Timestamp::from_str_utc(&String::from_utf8_lossy(b))
                    .map_err(|e| {
                        error::MysqlValueConversionSnafu {
                            err_msg: e.to_string(),
                        }
                        .build()
                    })?
                    .convert_to(ts_type.unit())
                    .ok_or_else(|| {
                        error::MysqlValueConversionSnafu {
                            err_msg: "Overflow when converting timestamp to target unit"
                                .to_string(),
                        }
                        .build()
                    })?;
```

Comment (sunng87) [reply src/servers/src/mysql/helper.rs:184]: sounds good. we can add some unit tests to verify this.

**Review by sunng87 — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**